library/Icingadb/Authentication/ObjectAuthorization.php

```diff
@@ -208,4 +216,37 @@ protected function checkGrants($permission, $roles)
 
         return false;
     }
+
+    /**
+     * Check whether the given filter matches the given object
+     *
+     * @param string $restriction
+     * @param Model  $object
+     *
+     * @return bool
+     */
+    public static function matchesOn($restriction, Model $object)
+    {
+        $self = new static();
+
+        $tableName = $object->getTableName();
+        $uniqueId = $object->{$object->getKeyName()};
+        if (! isset(self::$matchedFilters[$tableName][$uniqueId])) {
+            $restriction = QueryString::fromString($restriction)->parse();
```

Comment:
Why not `$this->parseRestriction()`? If because you don't have the restriction name, identify that based on the object type.

Reply:
Because they are validated anyway when creating the navigation itms. I don't have to parse them again.

Reply:
There may have something changed the filter besides the form. Don't trust that the user doesn't bypass the form validation somehow.